### PR TITLE
Check if school taught CS based on the latest census

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds a new scholarship requirement that checks that the teacher's school does not teach computer science according to the last access report. The `CensusSummaries` table has that information and is checked when calling `auto_score!`.

## Links
- [spec](https://docs.google.com/document/d/1q10AMvXibukiMjTwe_sRSgBbH6B4hjUHGs-d_dF_38s/edit)
- [rubric](https://docs.google.com/document/d/1vMGt2u1FExWKjJv9sp3mm7yyp7LalSS4OnRs_tUnEOw/edit#heading=h.sugthxc76yp1)
- [jira](https://codedotorg.atlassian.net/browse/ACQ-371)

### CODE REVIEW

**Comment by bethanyaconnor** [comment]
Note: the unit test Drone run is green but the UI tests are failing for everyone. It's possible there is a genuine failing UI test, though I think it's unlikely. I'll be sure to get a green Drone run before merging.

File: dashboard/app/models/pd/application/teacher_application.rb
Comment (TurnerRiley) [inline dashboard/app/models/pd/application/teacher_application.rb]: Nit: update the comment to say it lags by 1 year since you subtract 1 from the first application year

File: dashboard/app/models/census/census_summary.rb
Comment (TurnerRiley) [inline dashboard/app/models/census/census_summary.rb:88]: How does this check that the `teaches_cs` field is "YES" or "HISTORICAL_YES" ?

File: dashboard/app/models/pd/application/teacher_application.rb
Comment (TurnerRiley) [inline dashboard/app/models/pd/application/teacher_application.rb]: Nit: I think ruby safe chaining could work here since if `census_summary` is `nil` it will return `nil`. So I think it could be simplified to `census_summary.does_teach? ? NO : YES`

Comment (TurnerRiley) [reply dashboard/app/models/pd/application/teacher_application.rb]: Also just for my undertsanding, the scholarship criteria returns a `YES` if the school is NOT teaching CS? Just want to check because I got a bit confused with `does_teach?` matching to `NO` and vice versa. If this is the case, would you mind leaving a comment here explaining that please? I think I just found myself get caught up on the double negatives

File: dashboard/test/models/pd/application/teacher_application_test.rb
Comment (TurnerRiley) [inline dashboard/test/models/pd/application/teacher_application_test.rb]: Does the `not_teaching_in_access_report` rely on the principal approval? Or why is it `nil` in this case?

File: dashboard/app/models/census/census_summary.rb
Comment (bethanyaconnor) [reply dashboard/app/models/census/census_summary.rb:88]: It's a Rails thing, particularly `enum`. Here's some [documentation](https://api.rubyonrails.org/v5.1/classes/ActiveRecord/Enum.html) about it

File: dashboard/app/models/pd/application/teacher_application.rb
Comment (bethanyaconnor) [reply dashboard/app/models/pd/application/teacher_application.rb]: valid -- will add! And I got clarification from Dani that if the school isn't in the database, we should treat that as a "no" so I'll update this logic a bit too

Comment (bethanyaconnor) [reply dashboard/app/models/pd/application/teacher_application.rb]: I'm definitely open to better naming or something here, but I wanted to keep it consistent with other criteria that `YES` means they meet the criteria a scholarship and `NO` essentially means to defer to another criteria

File: dashboard/test/models/pd/application/teacher_application_test.rb
Comment (bethanyaconnor) [reply dashboard/test/models/pd/application/teacher_application_test.rb]: It was nil because there wasn't census_summaries data, but I'm changing that logic slightly so I'll update this!

File: dashboard/app/models/census/census_summary.rb
Comment (TurnerRiley) [reply dashboard/app/models/census/census_summary.rb:88]: Woah! TIL!

File: dashboard/app/models/pd/application/teacher_application.rb
Comment (TurnerRiley) [inline dashboard/app/models/pd/application/teacher_application.rb]: Tiny nit (not worth a drone rebuild if you're not already doing another one): duplicate word of "year years" and also "2023-2024" instead of "2023-2025"

**Review by TurnerRiley — APPROVED** [review]
Looks great! Just had one tiny nit that's not worth a rebuild unless you're already doing one

File: dashboard/app/models/pd/application/teacher_application.rb
Comment (TurnerRiley) [reply dashboard/app/models/pd/application/teacher_application.rb]: Yes totally makes sense! With your comment too I think it works great!

File: dashboard/app/models/census/census_summary.rb
Comment (bethanyaconnor) [reply dashboard/app/models/census/census_summary.rb:88]: haha I learned about it last week 😁

**Pull request merged**
